jep/0000/README.adoc

```diff
@@ -0,0 +1,345 @@
+* https://github.com/jenkinsci/script-scm-plugin SECURITY-461
+* https://github.com/jenkinsci/rtc-plugin (superseded by team-concert)
+
+
+===== Never released plugins
+
+Some plugins were never released.
+These will not be updated as part of this effort.
+
+* https://github.com/jenkinsci/cocoemma-plugin
+* https://github.com/jenkinsci/jwsdp-sqe-plugin
+* https://github.com/jenkinsci/pucm-plugin
+* https://github.com/jenkinsci/purecm-plugin
+
+
+== Motivation
+
+The link:https://commons.apache.org/proper/commons-digester/[Apache Commons Digester] that was included as a dependency of Jenkins Core is old and poorly maintained.
+It contains security vulnerabilities and also is not actually used by any part of Jenkins Core itself.
+Jenkins Core exposes an API to plugin that mitigates some of these security issues, but it is not the job of Jenkins to do this.
+Instead, we should remove the Digester dependency from Jenkins Core and the plugins that actually use Digester should handle these security concerns themselves.
+Also, because it is a dependency of Core, some plugins have taken a dependency on Digester being provided by Jenkins rather than using the API provided by Jenkins.
+
+Plugins in either of these scenarios will need to be updated or they will break after this removal.
+
+== Reasoning
+
+=== Risks of complete removal
+
+We chose to remove Digester from Jenkins Core in a manner that causes plugins which depend on it to fail unless they are updated.
+We made this choice based on an analysis of all plugins which are currently distributed by the Jenkins Update Center or which are
+in a repository in the link:https://github.com/jenkinsci[`jenkinsci` org on GitHub].
+All but a few of the plugins affected by this change are some combination of unmaintained, rarely used, and no longer or never published.
+We submitted PRs to all affected plugins.
+
+
+=== Detached plugin to maintain compatibility
+
+The current plan causes plugins which depend on Jenkins to provide Digester to fail unless they are updated.
+This could be mitigated by moving this dependency to a detached plugin.
+We decided against creating a detached pluging because there were a small number of affected plugins and only a few of them have significant install base.
+The creating and maintaining of a detached plugin would still be a significant amount of work and would cause the security vulnerabilities we are trying to address to remain open.
+
+
+== Backwards Compatibility
+
+The current plan causes plugins which depend on Jenkins to provide Digester to fail unless they are updated.
+The majority of this JEP is devoted to discussion of how to mitigate these breaks.
+
+
+== Security
+
+The motivation for this change is mitigating a number of known security issues
+
+== Infrastructure Requirements
+
+There are no new infrastructure requirements related to this proposal.
+
+== Testing
+
+There are no testing issues related to this proposal.
```

Comment:
`plugin-compat-tester` will need to be run on a reasonably large set of plugins to check for regressions introduced by the proposed core change.

Reply:
The proposed core change removes `commons-digester` and the `Digester2` API.  We've done a search for references to those in plugins in the `jenkinsci` and have filed PRs to address those.  What would you expect the `plugin-compat-tester` run to catch that a straight code search hasn't?

Reply:
OK, if you are confident everything that might break would be caught by existing code searches, feel free to mark this & https://github.com/jenkinsci/jep/pull/361#discussion_r629703813 resolved.

Reply:
@jglick @timja 
I'm confident, but I'm willing to acknowledge I may be missing something.  

What would you expect the `plugin-compat-tester` run to catch that a straight code search hasn't?
What would it take to do a PCT run?

Regardless of the result of this conversation, I'll note this in the Reasoning section.

Reply:
> What would you expect the `plugin-compat-tester` run to catch that a straight code search hasn't?

Nothing that I can think of in this case. In most core patches we would run PCT.

> What would it take to do a PCT run?

CloudBees has facilities to do so but the Jenkins project currently does not, alas.